### PR TITLE
Add dynamic way to choose controller scheme

### LINKED ISSUE #8: Add dynamic way to choose controller scheme
Inputs: Controller name/type
Output: Controller button and joystick mappings

Package: control
Node: teleop_control

Different controllers have different mappings to the buttons and axes arrays in the [joy msg](http://docs.ros.org/jade/api/sensor_msgs/html/msg/Joy.html) and to make it easier to switch between controllers it would be nice to have a dynamic way to switch these.  One option might be ros parameters but the only requirement is that it be simple to change between mappings such as the ones seen [here](http://wiki.ros.org/joy).

### PR DESCRIPTION
**Pull request by aswgxf**
Adds an easier way to access data from the parameter server, primarily controller mappings at this point. Closes #8

### CODE REVIEW

**Comment by aswgxf** [comment]
@matthewia94 @islamelnabarawy

File: ros_ws/src/common/config.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+#ifndef CONFIG_H_
+#define CONFIG_H_
+
+
+#include <ros/ros.h>
+#include "constants.h"
+
+
+struct ConfigValues
```
Comment (matthewia94) [inline ros_ws/src/common/config.h:20]: This struct may end up getting really big when everything is said and done.  A hash table or map might be a better solution.  @islamelnabarawy What do you think?

File: ros_ws/src/control/launch/robot.launch [hunk 2]
```diff
@@ -16,6 +16,7 @@
     
     <!-- Load parameter file -->
     <rosparam command="load" file="$(find control)/param/igvc.yaml"/>
+    <rosparam command="load" file="$(find control)/param/controllers/moga.yaml"/>
```
Comment (matthewia94) [inline ros_ws/src/control/launch/robot.launch]: It would be nice if we could pick which controller to use with a command line argument.  We could use the xbox as the default but use a different controller file if they pass it when launching.  More info here: http://wiki.ros.org/roslaunch/XML/arg

File: ros_ws/src/control/src/control/remote_control.cpp [hunk 3]
```diff
@@ -17,7 +17,7 @@ RemoteControl::RemoteControl()
     cmd_vel.angular.y = 0;
     cmd_vel.angular.z = 0;
     speed_modifier = 0.5;
-
+    config = initConfigs();
```
Comment (matthewia94) [inline ros_ws/src/control/src/control/remote_control.cpp:22]: I would separate this into its own code block to distinguish it from the rest of the initialization.  It might also be helpful if you left a comment describing what it does here so its not necessary to go looking for its implementation later.

File: ros_ws/src/control/src/control/twist_mux.cpp [hunk 4]
```diff
@@ -11,7 +11,8 @@ TwistMux::TwistMux()
     //Ensure that the robot starts in a safe standby mode
     current_mode = Control::standby;
     stopRobot();
-
+    //Get remote values
+    config = initConfigs();
```
Comment (matthewia94) [inline ros_ws/src/control/src/control/twist_mux.cpp:16]: Once again I would separate this into blocks.

File: ros_ws/src/vision/CMakeLists.txt [hunk 5]
```diff
@@ -13,6 +13,7 @@ find_package(catkin REQUIRED COMPONENTS
 ## catkin specific configuration ##
 ###################################
 catkin_package(
```
Comment (matthewia94) [inline ros_ws/src/vision/CMakeLists.txt]: This should not have been committed as it was fixed in #69.  Please check that you aren't adding unrelated files when adding to your commit.

File: ros_ws/src/control/src/control/remote_control.cpp (hunk 3 above)
Comment (aswgxf) [inline ros_ws/src/control/src/control/remote_control.cpp:22]: Does it need to be added to  RemoteControl::RemoteControl(std::string twist_topic, std::string joy_topic) ?

Comment (matthewia94) [inline ros_ws/src/control/src/control/remote_control.cpp:22]: Yes, that would be helpful since that constructor works the same way

File: ros_ws/src/common/config.h (hunk 1 above)
Comment (aswgxf) [inline ros_ws/src/common/config.h:20]: I'm not sure what those are/how to do that, but I could take a look if it'd be better.

Comment (matthewia94) [inline ros_ws/src/common/config.h:20]: I'm not sure if it is better or not, lets wait and see what Sam thinks.

**Comment by matthewia94** [comment]
@islamelnabarawy I looked over this and everything seems fine to me.  Would you mind taking a look to make sure that the structure looks okay to you.

**Comment by islamelnabarawy** [comment]
@matthewia94 
I agree, the structure would probably end up getting pretty large. There will only ever be a single object of it at any time though, correct? Also, it'll occupy the same size whether we use a dynamic structure or a static one, since all of those configurations have values, right?

If it's true that there will always be exactly one single object of this struct at any time, it might be a good idea to convert it to a singleton pattern with a private constructor. I can elaborate on exactly how that would work if you need me to. I wouldn't do it for this pull request, though; I'd create a separate task for it.

File: ros_ws/src/control/param/controllers/moga.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+controller_type:   moga
+
+lin_vel_axis:      1
+ang_vel_axis:      2
+
+teleop_btn:        0
+standby_btn:       1
+
+autonomous_btn:    3
+speed_inc_btn:     4
```
Comment (matthewia94) [inline ros_ws/src/control/param/controllers/moga.yaml]: These are reversed

Comment (aswgxf) [inline ros_ws/src/control/param/controllers/moga.yaml]: Should be fixed now

**Comment by aswgxf** [comment]
@matthewia94
